gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -554,28 +555,21 @@ public EntityFetcherResponse getTimeAggregatedMetrics(
   }
 
   @Override
-  public int getTotalEntities(EntitiesRequestContext requestContext, EntitiesRequest entitiesRequest) {
+  public Set<EntityKey> getTotalEntities(EntitiesRequestContext requestContext, EntitiesRequest entitiesRequest) {
```

Comment:
Is there a point in retaining this method when we are anyways relying on getEntities. Should we change the callers itself to use getEntities instead?

Reply:
This gives us a flexibility to modify the total entities request. For example: QueryServiceEntityFetcher is setting a limit of 10k on the total entities request.

Reply:
makes sense. done